cs130_front/src/components/views/GroupsPage/GroupsPage.js

```diff
@@ -1,35 +1,104 @@
 import React, { useState, useEffect } from 'react';
+import ClassList from '../../CurrentClasses/ClassList';
+import GroupItem from './GroupItem';
+import Text from '../../Text/Text';
+import Button from '../../Button/Button';
+import Requests from '../../Requests/Requests';
+import * as Colors from '../../../constants/Colors';
 import UserList from '../../UserList/UserList';
 import '../../../App.css';
+import CreateGroup from './CreateGroup';
 
 function GroupsPage(props) {
 
 
-  const classes = [
-    {name: "Discrete Mathematics", courseId: 1, groups: [{name: "DM Squad", groupId: 1}, {name: "DM Squad II", groupId: 1}] }, 
-    {name: "Computer Architecture", courseId: 2, groups: [{name: "CA Squad", groupId: 3}]}
+  const classes2 = [
+    { name: "Discrete Mathematics", courseId: 1, groups: [{ name: "DM Squad", groupId: 1 }, { name: "DM Squad II", groupId: 1 }] },
+    { name: "Computer Architecture", courseId: 2, groups: [{ name: "CA Squad", groupId: 3 }] }
   ]
-  const [mainPanelState, setMainPanel] = useState();
 
-  const members= [
-    {name: "Shirly fang", school: "UCLA", id:123, discord:"shirly#123", email:"shirly@gmail.com"},
-    {name: "Shirly fang", id:123, discord:"shirly#123", email:"shirly@gmail.com"}
+  const members = [
+    { name: "Shirly fang", school: "UCLA", id: 123, discord: "shirly#123", email: "shirly@gmail.com" },
+    { name: "Shirly fang", id: 123, discord: "shirly#123", email: "shirly@gmail.com" }
   ]
+
+  const groupInformation = [
+    { name: "Group name", value: "" },
+  ];
+
+  const reqs = [{ name: "Al Squad", id: 223, types: "invitation" }, { name: "Calc Gang", id: 223, types: "invitation" }]
+
+  const [requests, setRequests] = useState(reqs);
+  const [mainPanelState, setMainPanel] = useState();
+  const [currentGroup, setCurrentGroup] = useState({ name: "DM Squad", groupId: 1 });
+  const [classes, setClasses] = useState(classes2);
+
   // Pass this to ClassList
   function groupClicked(title) {
     //set main content to be for title
     setMainPanel(title);
+    setCurrentGroup();
```

Comment:
Finish this logic. if you have an array of groups, then setCurrentGroup(groups[onethathasthegroupname]). you have to loop thru groups and find the match.

Reply:
Is title correct for setMainPanel though?

Reply:
Or do I pass in group to both of them?

Reply:
When I pass in group to both it compiles and looks good

Reply:
so we dont need setMainPanel at all i think, only setCurrentGroup. this is b/c mainpanel can only be a list of users and groupname/meeting time info while for profile/classes theres a few different things. So, we only need to change the group being displayed. u only need the default case in ur switch statement when you render a groupitem.